### PR TITLE
Embed: Fix failure placeholder alignment/sizing

### LINKED ISSUE #19662: Embed failure placeholder design is broken
Capture d’écran 2020-01-15 à 11 19 11 AM

I believe this should look closer to the embed default placeholder with the failure message and buttons under the input?

Capture d’écran 2020-01-15 à 11 20 15 AM

I guess this happened with the placeholder refactorings cc @jasmussen @ItsJonQ

### PR DESCRIPTION
**Pull request by ItsJonQ**
## Description
This update fixes the positioning, alignment, and sizing of the error message that appears in the `Embed` block, specifically when there is an error.

## How has this been tested?
* Tested locally
* Deliberately caused an error in Embedding to cause the message to show
* Ensure the UI looks correct for Embed URL, as well as other Embed placeholders

## Screenshots

Screen Shot 2020-01-15 at 10 06 25 AM

In the above screenshot, we can see that the placeholder sizing/alignment appear correctly for regular embeds, table embeds, and embeds with errors.

## Types of changes
* Adjusted HTML markup structure of the Embed placeholder in order to resolve issue caused by `display: flex` wrapper
* Adjusted CSS to ensure proper sizing/alignment
* Replaced the `p` tag with a `div` to avoid inheriting `p` styles from the editor. I felt like this was okay, considering the instructions HTML selector was also a `div`

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. .

Resolves https://github.com/WordPress/gutenberg/issues/19662

### CODE REVIEW

File: packages/block-library/src/table/editor.scss [hunk 1]
```diff
@@ -63,6 +63,10 @@
 		}
 	}
 
+	.components-base-control__field {
+		margin-bottom: 0;
```
Comment (ItsJonQ) [inline packages/block-library/src/table/editor.scss]: This removes the bottom margins of the input fields, allowing the Inputs + Button to align correctly.

File: packages/components/src/placeholder/style.scss [hunk 2]
```diff
@@ -74,6 +75,11 @@
 	margin-bottom: 1em;
 }
 
+.components-placeholder__error {
+	margin-top: 1em;
+	width: 100%;
```
Comment (ItsJonQ) [inline packages/components/src/placeholder/style.scss]: Required to render underneath the help text, considering the `display: flex` based wrapper.

**Review by youknowriad — APPROVED** [review]
Maybe the error text is too close to the buttons;

Thanks a lot for the fix, this is way better.

**Comment by ItsJonQ** [comment]
@youknowriad Thanks for the feedback! I've made that adjustment:

Screen Shot 2020-01-21 at 4 38 09 PM

Once Travis is green and happy, I'll merge it up :)

File: packages/block-library/src/embed/embed-placeholder.js [hunk 3]
```diff
@@ -28,18 +28,20 @@ const EmbedPlaceholder = ( props ) => {
...
-				}
 			</form>
 			<div className="components-placeholder__learn-more">
 				<ExternalLink href={ __( 'https://wordpress.org/support/article/embeds/' ) }>
 					{ __( 'Learn more about embeds' ) }
 				</ExternalLink>
 			</div>
+			{ cannotEmbed &&
+				<div className="components-placeholder__error">
+					<div className="components-placeholder__instructions">
```
Comment (aduth) [inline packages/block-library/src/embed/embed-placeholder.js]: I see that the precedent existed previously, but this change contributed further to it:

`components-placeholder__instructions` is not a valid class name to use in this context.

>A component's class name should never be used outside its own folder (with rare exceptions such as _z-index.scss). If you need to inherit styles of another component in your own components, you should render an instance of that other component. At worst, you should duplicate the styles within your own component's stylesheet. This is intended to improve maintainability by isolating shared components as a reusable interface, reducing the surface area of similar UI elements by adapting a limited set of common components to support a varied set of use-cases.

https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#naming

**Pull request merged**